### PR TITLE
Nikita / User Preferred language 

### PR DESCRIPTION
**Pull request by nikita-deriv**
User got site with preferred language, after he logs in.

### CODE REVIEW

File: src/javascript/_common/base/client_base.js [hunk 1]
```diff
@@ -37,6 +37,9 @@ const ClientBase = (() => {
      * @param {String|null} loginid        The account to set the value for
      */
     const set = (key, value, loginid = current_loginid) => {
+        if (key === 'preferred_language' &&  !LocalStore.get('preferred_language')) {
```
Comment (akmal-binary) [inline src/javascript/_common/base/client_base.js:40]:
```suggestion
        if (key === 'preferred_language' && !LocalStore.get('preferred_language')) {
```

File: src/javascript/_common/base/login.js [hunk 2]
```diff
@@ -8,20 +8,21 @@ const removeCookies       = require('../storage').removeCookies;
 const urlForCurrentDomain = require('../url').urlForCurrentDomain;
 const isLoginPages        = require('../utility').isLoginPages;
 const TrafficSource       = require('../../app/common/traffic_source');
+const urlFor = require('../language').urlFor;
```
Comment (akmal-binary) [inline src/javascript/_common/base/login.js:11]: Could you add more spaces before equals sign to match it with other equal signs above?

File: src/javascript/_common/base/login.js [hunk 3]
```diff
@@ -8,20 +8,21 @@ const removeCookies       = require('../storage').removeCookies;
...
-            sessionStorage.setItem('redirect_url', window.location.href);
+            sessionStorage.setItem('redirect_url', urlFor(LocalStore.get('preferred_language'), window.location.href));
             window.location.href = loginUrl();
         }
     };
 
     const loginUrl = () => {
         const server_url = localStorage.getItem('config.server_url');
-        const language   = getLanguage();
+        const language = LocalStore.get('preferred_language') || getLanguage();
```
Comment (akmal-binary) [inline src/javascript/_common/base/login.js:25]: Also, add spaces here and above line to match with below lines

File: src/javascript/app/common/attach_dom/language_dropdown.js [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
+const LocalStore = require('../../../_common/storage').LocalStore;
 
 const createLanguageDropDown = (website_status) => {
     const $languages = $('.languages');
 
     if ($languages.find('#display_language li').attr('class')) {
         return;
     }
-
+
```
Comment (akmal-binary) [inline src/javascript/app/common/attach_dom/language_dropdown.js:10]: Remove this tab please.

File: src/javascript/_common/base/client_base.js (hunk 1 above)
Comment (nikita-deriv) [reply src/javascript/_common/base/client_base.js]: Removed additional space here.

File: src/javascript/_common/base/login.js (hunk 2 above)
Comment (nikita-deriv) [reply src/javascript/_common/base/login.js]: Thanks for noticing, forget about that one. Changed.

File: src/javascript/_common/base/login.js (hunk 3 above)
Comment (nikita-deriv) [reply src/javascript/_common/base/login.js]: Made equations in one line.

File: src/javascript/app/common/attach_dom/language_dropdown.js (hunk 4 above)
Comment (nikita-deriv) [reply src/javascript/app/common/attach_dom/language_dropdown.js]: Removed.

**Review by akmal-binary — APPROVED** [review]

**Review by akmal-binary — APPROVED** [review]

**Pull request merged**